Mage/Module.cs

```diff
@@ -8,29 +8,45 @@
 
 namespace Wizcorp.MageSDK.MageClient
 {
-
-	public class Module<T> : Singleton<T> where T : class, new()
+	using MageCommandAction = Action<JObject, Action<Exception, JToken>>;
+	using MageCommandFunction = Func<JObject, UserCommandStatus>;
+	
+	public abstract class Module<T> : Singleton<T> where T : class, new()	
 	{
-		//
+		// Tells us if the module was set up
+		private bool _SetupCompleted = false;
+		public bool SetupCompleted { 
+			get { return _SetupCompleted; }
+			private set { _SetupCompleted = value; }
+		}
+		
+
+		// Mage singleton accessor
 		protected Mage Mage
 		{
 			get { return Mage.Instance; }
 		}
 
+		// Contextualized logger
 		protected Logger Logger
 		{
 			get { return Mage.Logger(GetType().Name); }
 		}
 
-
-		//
+		// List of static topics to load during setup
 		protected virtual List<string> StaticTopics
 		{
-			get { return null; }
+			get { return new List<string> {}; }
```

Comment:
For this we have to be sure we catch all of the early returns and any that are actually continued expecting something to have been there. If there is nothing to do our methods should handle either null or empty list.

I would argue you are going to have less errors if you have empty list, but you have to make sure you check this in your functions.